modules/pricer/src/main/java/com/opengamma/strata/pricer/sensitivity/CurveSensitivityUtils.java

```diff
@@ -0,0 +1,160 @@
+   * 
+   * @param sensitivities  the input sensitivities
+   * @param targetDates  the list of dates for the re-bucketing
+   * @param sensitivityDate  the date for which the sensitivities are valid
+   * @return the sensitivity after the re-bucketing
+   */
+  public static CurveCurrencyParameterSensitivities linearRebucketing(
+      CurveCurrencyParameterSensitivities sensitivities,
+      List<LocalDate> targetDates,
+      LocalDate sensitivityDate) {
+    int nbBuckets = targetDates.size();
+    List<CurveParameterMetadata> pmdTarget = new ArrayList<>();
+    for (int i = 0; i < nbBuckets; i++) {
+      pmdTarget.add(SimpleCurveNodeMetadata.of(targetDates.get(i), targetDates.get(i).toString()));
+    }
+    ImmutableList<CurveCurrencyParameterSensitivity> listSensi = sensitivities.getSensitivities();
+    List<CurveCurrencyParameterSensitivity> sensitivityTarget = new ArrayList<>();
+    for (CurveCurrencyParameterSensitivity s : listSensi) {
+      double[] rebucketedSensitivityAmounts = new double[nbBuckets];
+      CurveMetadata m = s.getMetadata();
+      DoubleArray sa = s.getSensitivity();
+      ArgChecker.isTrue(m.getParameterMetadata().isPresent(), "parameter metadata must be present");
+      List<CurveParameterMetadata> lm = m.getParameterMetadata().get();
+      for (int loopnode = 0; loopnode < sa.size(); loopnode++) {
+        ArgChecker.isTrue((lm.get(loopnode) instanceof DatedCurveParameterMetadata) ||
+            (lm.get(loopnode) instanceof TenorCurveNodeMetadata), "re-bucketing require sensitivity date");
+        LocalDate nodeDate;
+        if (lm.get(loopnode) instanceof DatedCurveParameterMetadata) {
+          DatedCurveParameterMetadata dpm = (DatedCurveParameterMetadata) lm.get(loopnode);
+          nodeDate = dpm.getDate();
+        } else {
+          TenorCurveNodeMetadata tpm = (TenorCurveNodeMetadata) lm.get(loopnode);
+          nodeDate = sensitivityDate.plus(tpm.getTenor());
+        }
+        rebucketingArray(targetDates, nbBuckets, rebucketedSensitivityAmounts, sa.get(loopnode), nodeDate);
+      }
+      CurveCurrencyParameterSensitivity rebucketedSensitivity =
+          CurveCurrencyParameterSensitivity.of(
+              DefaultCurveMetadata.builder().curveName(s.getCurveName()).parameterMetadata(pmdTarget).build(),
+              s.getCurrency(), DoubleArray.ofUnsafe(rebucketedSensitivityAmounts));
+      sensitivityTarget.add(rebucketedSensitivity);
+    }
+    return CurveCurrencyParameterSensitivities.of(sensitivityTarget);
+  }
+
+  /**
+   * 
+   * @param targetDates
+   * @param nbBuckets
+   * @param rebucketedSensitivityAmounts  the array of sensitivities; the array is modified by the method
+   * @param sa
+   * @param nodeDate  the date associated to the node re-bucketed
+   */
+  private static void rebucketingArray(
+      List<LocalDate> targetDates, 
+      int nbBuckets, 
+      double[] rebucketedSensitivityAmounts, 
```

Comment:
Would it be possible to structure this method to return a single sensitivity value? The parameters would be:

* List of original node dates
* Sensitivities
* Date of a rebucketed node

The return value would be the sensitivity to the rebucketed node. This method could be called once for each of the rebucketed nodes and there would be no need to mutate the parameter.